src/backends/r1cs/mod.rs

```diff
@@ -310,6 +311,21 @@ where
         builtin::poseidon::<F>
     }
 
+    fn init_circuit(&mut self, func: &mut FunctionDef) {
+        // reorder the arguments by public and private
+        // the public inputs should be at the beginning
+        func.sig.arguments.sort_by_key(|arg| match &arg.attribute {
+            Some(attr) => match attr.kind {
+                crate::parser::types::AttributeKind::Pub => 0,
+                _ => 1,
+            },
+            None => 1,
+        });
```

Comment:
imo this should happen outside of this function, unless it becomes a backend specific thing for now this code could leave outside

Reply:
[99f328f](https://github.com/zksecurity/noname/pull/78/commits/99f328fa9d6799148eff5dd65a699758aeeceebe)